### ISSUE TITLE
Can't connect to Adafruit IO with FONA808

### ISSUE THREAD

**Issue by marcoschwartz** [adafruit/io-issues#65]
Hello,

I am having a problem with the FONA808 breakout with Adafruit IO. When connecting to Adafruit IO (using the example MQTT sketch), I have the following error:

AT+CIPSTART="TCP","io.adafruit.com","1883"
	<--- OK
	<--- STATE: PDP DEACT
Connection failed
Retrying MQTT connection in 5 seconds...
	---> AT+CIPSTATUS
	<--- OK
	<--- STATE: PDP DEACT
	---> AT+CIPCLOSE
	<--- +CME ERROR: operation not allowed

It's quite strange as otherwise the GPRS connection works well. I can for example grab the content of any web page, like the example Adafruit page from the FONAtest sketch. Do you know what could be the reason? Thanks!

**Comment by bborncr** [comment]
Did you ever get a response to this question? I am having the same problem.

**Comment by marcoschwartz** [comment]
No, but I still need an answer as well

**Comment by ladyada** [comment]
er, marco, didn't you write a tutorial about this? :)
https://learn.adafruit.com/geofencing-with-the-fona-808-and-adafruit-io/integration-with-adafruit-io

**Comment by marcoschwartz** [comment]
I had to use a SIM card from another carrier for the tutorial :) I used a SIM card that I am using a data only with my iPad. With the usual SIM card I was using before it still didn't work, so maybe it's an issue linked to some carriers or SIM cards

**Comment by Ngsnet** [comment]
I would try call carrier and ask them if they support all range of ports. I experienced troubles with one of our local provider a year ago or so. They offered internet but they were blocking some ports i tried to connect over GPRS.

Denis

**Comment by jjaiprakash** [comment]
Is this issue resolved? I have the same problem. I contacted my carrier. Their opinion is that prior to AT+CIPSTART there is a AT+CIPSHUT command which deactivates the GPRS PDP context. 
This has been the core part of Adafruit_FONA library TCPconnect function right from the beginning. My understanding is that this clears out existing connections before creating new ones. But my carrier disagrees. 
Any update would be greatly appreciated.
My log is included below (similar to Marco's log, but with couple additional commands).

Thanks.
Jai
Connecting to MQTT... .---> AT+CIPSHUT
.<--- SHUT OK
.---> AT+CIPMUX=0
.<--- OK
.---> AT+CIPRXGET=1
.<--- OK
AT+CIPSTART="TCP","io.adafruit.com","1883"
.<--- OK
.<--- STATE: PDP DEACT
Connection failed
Retrying MQTT connection in 5 seconds...

**Comment by ladyada** [comment]
@jjaiprakash you can try manual AT commands, if you find a path that works let us know. the command set we have works great with TMobile Ting...maybe other carriers are different?

**Comment by haku15** [comment]
Hi Marco!
I had the same problem. I posted a possible solution here [http://forums.adafruit.com/viewtopic.php?f=56&t=89410](url).
I hope it helps you, because you helped me a lot with your great tutorials :)

**Comment by ladyada** [comment]
FONA library was recently updated to have a more thorough TCP connection protocol, have you tried the latest version, especially since the "add AT+CSTT & AT+CIICR to enableGPRS" commit from jan17?

**Comment by haku15** [comment]
Hi Ladyada! I am using the Fona library downloaded on 1st Jan 2016. I haven't noticed you updated the it 17 days ago. Thanks for telling me and for the constant work on the library. Anyway I learned a lot by trying to resolve the problem  :)

**Comment by ladyada** [comment]
we all learned a lot! :) can you try the the latest library and let me know if it fixes it? then i could close the issue!

**Comment by haku15** [comment]
Sure! I'll try it in the next days and let you know the results.

**Comment by bborncr** [comment]
This fixed the issue with the ICE. Now the FONA works with all 3 service
providers here in Costa Rica. Thanks!

BTW, the FONA had a CRC error in "Manage LIbraries" and didn't allow the
latest library to install. I manually updated and everything worked great.

On Sat, Feb 6, 2016 at 6:55 PM, haku15 <notifications@github.com> wrote:

> Sure! I'll try it in the next days and let you know the results.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adafruit/io-issues/issues/65#issuecomment-180902520>.
>



-- 
Bentley Born
CRCibernetica.com
Tel. (506) 2258-5757 x111
Cel. (506) 8896-7539
bentley@ <bentley@gridshield.net>crcibernetica.com

**Comment by haku15** [comment]
I've tested the latest FONA library release (committed on 20 Jan). 
It works, very good! I don't see "PDP DEACT" anymore: now I see "STATE: CONNECT OK" and that's correct.
I've tried  MQTT and REST and both work, I could post data to feeds successfully.
Thank you very much!

**Comment by ladyada** [comment]
a few people have said this works for them so im gonna close this issue! plz reopen if anyone has Qs or issues

**Comment by sunderj91** [comment]
Hey guys,

I am trying to connect FONA 808 in geofencing project, but unable to connect mqtt. I have tried the fixes as mentioned in the comments but no use. I have tried to get the value of ret variable in the project, and its value is -1. Everything seems to be ok, since I get STATE: Connect OK.

AT+CIPSTART="TCP","io.adafruit.com","1883"
	<--- OK
	<--- CONNECT OK
	---> AT+CIPSTATUS
	<--- OK
	<--- STATE: CONNECT OK
AT+CIPSEND=61
 0x10 0x3B 0x0 0x4 0x4D 0x51 0x54 0x54 0x4 0xFFFFFFC2 0x1 0x2C 0x0 0x2 0x2D 0x20 0x0 0x9 0x73 0x75 0x6E 0x64 0x65 0x72 0x6A 0x39 0x31 0x0 0x20 0x38 0x30 0x61 0x62 0x65 0x63 0x66 0x30 0x66 0x63 0x35 0x64 0x34 0x62 0x61 0x37 0x61 0x35 0x32 0x65 0x65 0x64 0x39 0x64 0x36 0x32 0x32 0x37 0x30 0x31 0x35 0x66
	<--- 
Connection failed
-1
Retrying MQTT connection in 5 seconds...
	---> AT+CIPSTATUS
	<--- 
	---> AT+CIPCLOSE
	<--- 
	---> AT+CIPSHUT
	<--- 

Please help me understand what could be the issue.

**Comment by durvesh09** [comment]
same problem..

**Comment by toddtreece** [comment]
@durvesh09 give this updated sketch a try: https://github.com/toddtreece/arduino-geofencing/blob/master/geo_fencing_io/geo_fencing_io.ino

**Comment by marimic** [comment]
Hi 
Has anyone tried to use the FONA 808 to push sensor data to a server - say like Bluehost? We've tried just testing with the FONAtest to send data. We were able to connect with our site but unsuccessful in sending data to our database (we've setup a database to store sensor data from 6 different FONA's). We are using Ting SIMs. This forum is old so just wondering if anyone has tried anything like this and might share some of their code and/or info whether this is even a possibility.

Thanks
Maria

**Comment by amason101070** [comment]
Hi Maria,

I'm actually looking to do the same thing so if I find out I will post it for you. If you find out it would be great if you let me know too.

Thanks!

Ron

**Comment by marimic** [comment]
Hi Ron

We finally got it to work a while back but it took some effort. We used the AT command to write to the table on our database. Most of the code is cut and pasted from Adafruit's test (ie. for gprs and gsm). The hard part was on the server side in php (I'm not the code writer so i can't really post it here). Here's what we did in the Arduino code to write to database:

uint16_t statuscode;
  int16_t length;
  if (isnan(h) ) {
    h = 0.0f;
  }
  if (isnan(f) ) {
    f = 0.0f;
  }

  char url[120] = "urlhere.com/write_data.php?val=";      //write to database
  strcat(url, floatToString(dChar, dustVal, 4));                   //dust sensor value
  strcat(url, "&lat=");
  strcat(url, floatToString(aChar, latitude, 12));                //gps location latitude
  strcat(url, "&lon=");
  strcat(url, floatToString(oChar, longitude, 12));           //gps location longitute (note: use "lon" in the 
                                                                                        php code)

  if (DEBUG) {
    Serial.println(url);
  }

  if (!fona.HTTP_GET_start(url, &statuscode, (uint16_t *)&length)) {
    Serial.println("Failed!");
  }
  else {
    isSent = true;
    Serial.println("Sent Ok");
  }

Hope this is helpful. As I said, much of the code is built around the FONAtest example.
My next objective is to have the data from the database map onto a google map, tracking the sensors in real time on our website.

Good luck with your project!

Maria
